python/pyspark/sql/dataframe.py

```diff
@@ -2423,6 +2423,43 @@ def freqItems(self, cols, support=None):
             support = 0.01
         return DataFrame(self._jdf.stat().freqItems(_to_seq(self._sc, cols), support), self.sql_ctx)
 
+    def with_columns(self, col_names, cols):
```

Comment:
Similar to this change in python, i think same change is needed at the DataFrame.R to expose with_columns in R also.

Reply:
Let's keep it consistent with Scala side for now. Note that camelCase doesn't violate PEP8 (e.g., `threading` in Python)

Reply:
@HyukjinKwon OK, I will change the name to `withColumns `

Reply:
@SaurabhChawla100 Thanks for reminder, I am not very familar with R, I am going to submit it in a separate PR.